x/stakeibc/keeper/unbonding_records.go

```diff
@@ -315,69 +318,75 @@ func (k Keeper) SweepAllUnbondedTokensForHostZone(ctx sdk.Context, hostZone type
-			msgs = append(msgs, &banktypes.MsgSend{
-				FromAddress: delegationAccount.GetAddress(),
-				ToAddress:   redemptionAccount.GetAddress(),
-				Amount:      sdk.NewCoins(sweepCoin),
-			})
-			k.Logger(ctx).Info(fmt.Sprintf("Bank sending unbonded tokens batch, from delegation to redemption account. Msg: %v", msgs))
-
-			// store the epoch numbers in the callback to identify the epoch unbonding records
-			redemptionCallback := types.RedemptionCallback{
-				HostZoneId:              hostZone.ChainId,
-				EpochUnbondingRecordIds: epochUnbondingRecordIds,
-			}
+	// If we have any amount to sweep, then we can send the ICA call to sweep them
+	if totalAmtTransferToRedemptionAcct == 0 {
+		k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId, "No unbonded tokens ready for sweep"))
+		return true, totalAmtTransferToRedemptionAcct
+	}
+	k.Logger(ctx).Info(utils.LogWithHostZone(hostZone.ChainId, "Batch transferring %d to host zone", totalAmtTransferToRedemptionAcct))
 
-			marshalledCallbackArgs, err := k.MarshalRedemptionCallbackArgs(ctx, redemptionCallback)
-			if err != nil {
-				k.Logger(ctx).Error(err.Error())
-				return false, 0
-			}
+	// Get the delegation account and redemption account
+	delegationAccount := hostZone.DelegationAccount
+	if delegationAccount == nil || delegationAccount.Address == "" {
+		k.Logger(ctx).Error(fmt.Sprintf("Zone %s is missing a delegation address!", hostZone.ChainId))
+		return false, 0
+	}
+	redemptionAccount := hostZone.RedemptionAccount
+	if redemptionAccount == nil || redemptionAccount.Address == "" {
+		k.Logger(ctx).Error(fmt.Sprintf("Zone %s is missing a redemption address!", hostZone.ChainId))
+		return false, 0
+	}
 
-			// Send the transaction through SubmitTx
-			_, err = k.SubmitTxsDayEpoch(ctx, hostZone.ConnectionId, msgs, *delegationAccount, ICACallbackID_Redemption, marshalledCallbackArgs)
-			if err != nil {
-				k.Logger(ctx).Info(fmt.Sprintf("Failed to SubmitTxs, transfer to redemption account on %s", hostZone.ChainId))
-			}
-			err = k.RecordsKeeper.SetHostZoneUnbondings(ctx, hostZone.ChainId, epochUnbondingRecordIds, recordstypes.HostZoneUnbonding_EXIT_TRANSFER_IN_PROGRESS)
-			if err != nil {
-				k.Logger(ctx).Error(err.Error())
-				return false, 0
-			}
-			k.Logger(ctx).Info(fmt.Sprintf("Successfully completed unbonded token sweep ICA call for %s, %s, %v", hostZone.ConnectionId, hostZone.ChainId, msgs))
-		} else {
-			k.Logger(ctx).Info(fmt.Sprintf("\tNot sweeping tokens for host zone %s because redemption/delegation accounts aren't registered", hostZone.ChainId))
-			return false, 0
-		}
-	} else {
-		k.Logger(ctx).Info(fmt.Sprintf("\tNo unbonded tokens this day to sweep for host zone %s", hostZone.ChainId))
+	// Build transfer message to transfer from the delegation account to redemption account
+	sweepCoin := sdk.NewCoin(hostZone.HostDenom, sdk.NewInt(totalAmtTransferToRedemptionAcct))
+	msgs := []sdk.Msg{
+		&banktypes.MsgSend{
+			FromAddress: delegationAccount.Address,
```

Comment:
Same Q as above - why removing the getters?

Reply:
chatted offline: TLDR, accessing fields directly is preferred over getters to prevent unintended consequences